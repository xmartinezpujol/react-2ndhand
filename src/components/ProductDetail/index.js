import React from 'react';
import { Link } from 'react-router-dom';

import {
  CardDescription,
  CardInfo,
  CardWrapper,
  DetailContainer,
  DetailPageWrapper,
  LabelAdapter,
  Title
} from "./styles";
// $FlowFixMe
import { Card, Icon } from "semantic-ui-react";

const ProductDetail = ({
  product,
}: {
  product: ProductType
}) => (
  <DetailPageWrapper>
    <DetailContainer>
      <Link to="/list" href="/list" data-cy="productCardDetailBackArrow">
        <Icon color='teal' name='arrow left' size='big' />
      </Link>
      <CardWrapper data-cy="productCardDetail">
        {product &&
          <React.Fragment>
            <Card.Content>
              <Card.Header>
                <Title>{product.title}</Title>
              </Card.Header>
              <Card.Meta>
                {product.params.map(param => (
                  <LabelAdapter key={`${product.id}-${param[0]}`} color="teal">{param[0]}
                    <LabelAdapter.Detail>{param[1]}</LabelAdapter.Detail>
                  </LabelAdapter>
                ))}
              </Card.Meta>
              <CardDescription>{product.description}</CardDescription>
            </Card.Content>
            <Card.Content extra>
              <Icon color="teal" name="time" />
              <CardInfo>{`${product.created}`}</CardInfo>
              <Icon color="teal" name="map marker" />
              <CardInfo>{`${product.city_label}`}</CardInfo>
            </Card.Content>
          </React.Fragment>
        }
        {!product &&
          <React.Fragment>
            <Card.Content>
              <Card.Header>
                <Title>Produkt nie znaleziony</Title>
              </Card.Header>
            </Card.Content>
          </React.Fragment>
        }
      </CardWrapper>
    </DetailContainer>
  </DetailPageWrapper>
);

export default ProductDetail;
