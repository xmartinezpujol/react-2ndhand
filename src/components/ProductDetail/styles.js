import styled from 'styled-components';
// $FlowFixMe
import { Card, Label } from 'semantic-ui-react';

export const DetailPageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
`;

export const DetailContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  width: 100%;
  padding: 40px 20px 20px 20px;
  @media (min-width: 768px) {
    padding: 50px 20px 20px 20px;
    width: calc(100% - 20px);
  }
  
  @media (min-width: 992px) {
    max-width: 1200px;
  }
`;

export const CardWrapper = styled(Card)`
  &&& {
    display: flex;
    margin-top: 60px;
    justify-content: center;
    min-height: 240px;
    width: 100%;
    
    @media (min-width: 768px) {
      width: calc(100% - 20px);
    }
    
    @media (min-width: 992px) {
      margin-top: 60px;
      width: 932px;
      padding: 20px;
    }
  }
`;

export const CardDescription = styled(Card.Description)`
  &&& {
    @media (min-width: 768px) {
      margin: 10px 0 20px 0;
    }
    
    @media (min-width: 992px) {
      font-size: 16px;
      text-align: justify;
      margin: 30px 0 20px 0 !important;
    }
  }
`;

export const CardInfo = styled.span`
  margin-right: 10px;
`;

export const LabelAdapter = styled(Label)`
  display: flex;
  font-size: 10px !important;
  
  @media (min-width: 768px) {
    font-size: 18px !important;
  }
  
  @media (min-width: 992px) {
    font-size: 14px !important;
  }
`;

export const Title = styled.div`
  font-size: 16px;
  margin-bottom: 10px;
  
  @media (min-width: 768px) {
    font-size: 18px;
  }
  
  @media (min-width: 992px) {
    font-size: 28px;
    margin: 10px 0 20px 0;
  }
`;
