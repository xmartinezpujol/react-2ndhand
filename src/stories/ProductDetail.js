import React from 'react';
import StoryRouter from 'storybook-react-router';

import { storiesOf } from '@storybook/react';

import { withKnobs } from '@storybook/addon-knobs';

import ProductDetail from '../components/ProductDetail';

storiesOf('ProductDetail', module)
  .addDecorator(withKnobs)
  .addDecorator(StoryRouter())
  .add('default', () => {
    const product = {
      "id": "205799123",
      "title": "iphone 5s 16gb silver bialy stan dbd Pl dystrybucja 100% sprawny",
      "created": "10  lut",
      "description": "Sprzedam Iphone 5s w bdb stanie technicznym i wizualnym (jak na foto) telefon jest w 100 % sprawny wszystko dziala nie zawiesza sie i nie gubi sieci. Wizualnie jak na zdjeciach posiada normalne slady urzytkowania,\nPl dystrybucja telefon fabrycznie bez blokady simlock. \nTelefon sprzedaje w kmpl \nW sklad zestawu wchodzi telefon ladowarka (ori) kabel sluchawki i opakowanie zgodne z imei",
      "params": [
        [
          "Marka",
          "iPhone"
        ],
        [
          "Stan",
          "Używane"
        ]
      ],
      "city_label": "Kobyłka"
    };
    return (
      <ProductDetail product={product} />
    );
  });
