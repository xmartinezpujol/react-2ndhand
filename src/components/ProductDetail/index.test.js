import React from 'react';
import { mount } from 'enzyme';
import { StaticRouter } from 'react-router';
import renderer from 'react-test-renderer';

// $FlowFixMe
import Ads from '../../mocks/ads.json';

import ProductDetail from './index';

describe('ProductDetail', () => {
  const addsByPage = Ads.map(page => page.ads);
  const product = [].concat.apply([], addsByPage)[0];

  it('renders correctly', () => {
    const context = {};
    const tree = renderer.create(
      <StaticRouter location="/list" context={context}>
        <ProductDetail product={product} />
      </StaticRouter>);
    const wrapper = mount(
      <StaticRouter location="/list" context={context}>
        <ProductDetail product={product} />
      </StaticRouter>);
    expect(tree).toMatchSnapshot();
    expect(wrapper.find('CardHeader').length).toBe(1);
    expect(wrapper.find('Label').length).toBe(2);
    expect(wrapper.find('CardMeta').length).toBe(1);
    expect(wrapper.find('CardDescription').length).toBe(1);
    expect(wrapper.find('Icon').length).toBe(3);
  });

  it('renders a message when data is empty', () => {
    const context = {};
    const product = undefined;
    const wrapper = mount(
      <StaticRouter location="/list" context={context}>
        <ProductDetail product={product} />
      </StaticRouter>);
    expect(wrapper.find('CardHeader').length).toBe(1);
    expect(wrapper.find('Label').length).toBe(0);
    expect(wrapper.find('CardMeta').length).toBe(0);
    expect(wrapper.find('CardDescription').length).toBe(0);
    expect(wrapper.find('Icon').length).toBe(1);
  });
});
