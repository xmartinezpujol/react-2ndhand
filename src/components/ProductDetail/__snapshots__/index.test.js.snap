// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail renders correctly 1`] = `
<div
  className="styles__DetailPageWrapper-sc-1ybtg9o-0 vGzzK"
>
  <div
    className="styles__DetailContainer-sc-1ybtg9o-1 kjWyKX"
  >
    <a
      href="/list"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="teal arrow left big icon"
        onClick={[Function]}
      />
    </a>
    <div
      className="ui card styles__CardWrapper-sc-1ybtg9o-2 irpKAG"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          <div
            className="styles__Title-sc-1ybtg9o-6 fRLAIq"
          >
            iphone 5s 16gb silver bialy stan dbd Pl dystrybucja 100% sprawny
          </div>
        </div>
        <div
          className="meta"
        >
          <div
            className="ui teal label styles__LabelAdapter-sc-1ybtg9o-5 ekGhay"
            onClick={[Function]}
          >
            Marka
            <div
              className="detail"
            >
              iPhone
            </div>
          </div>
          <div
            className="ui teal label styles__LabelAdapter-sc-1ybtg9o-5 ekGhay"
            onClick={[Function]}
          >
            Stan
            <div
              className="detail"
            >
              Używane
            </div>
          </div>
        </div>
        <div
          className="description styles__CardDescription-sc-1ybtg9o-3 esmgxQ"
        >
          Sprzedam Iphone 5s w bdb stanie technicznym i wizualnym (jak na foto) telefon jest w 100 % sprawny wszystko dziala nie zawiesza sie i nie gubi sieci. Wizualnie jak na zdjeciach posiada normalne slady urzytkowania,
Pl dystrybucja telefon fabrycznie bez blokady simlock. 
Telefon sprzedaje w kmpl 
W sklad zestawu wchodzi telefon ladowarka (ori) kabel sluchawki i opakowanie zgodne z imei
        </div>
      </div>
      <div
        className="extra content"
      >
        <i
          aria-hidden="true"
          className="teal time icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-1ybtg9o-4 ejvFdW"
        >
          10  lut
        </span>
        <i
          aria-hidden="true"
          className="teal map marker icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-1ybtg9o-4 ejvFdW"
        >
          Kobyłka
        </span>
      </div>
    </div>
  </div>
</div>
`;
